awx/api/generics.py

```diff
@@ -526,8 +529,14 @@ def get_description_context(self):
     def get_queryset(self):
         parent = self.get_parent_object()
         self.check_parent_access(parent)
-        qs = self.request.user.get_queryset(self.model).distinct()
         sublist_qs = self.get_sublist_queryset(parent)
+        if not self.filter_read_permission:
+            access_class = access_registry[self.model]
+            if access_class.prefetch_related:
+                return sublist_qs.prefetch_related(*access_class.prefetch_related)
+            if access_class.select_related:
+                return sublist_qs.select_related(*access_class.select_related)
```

Comment:
Do you want to be safe and raise an exception here that the access_class had neither prefetch_related nor select_related? Otherwise we turned off filter_read_permission but fall through to as if we hadn't.

Reply:
This is doing the same thing as a prior bit in this file. The argument for its robustness is that the `BaseAccess` class sets these to empty tuples.

https://github.com/ansible/awx/blob/7deddabea60166ca8808509756a080d63cffa97e/awx/main/access.py#L223